e2e/e2eutil/utils.go

```diff
@@ -78,3 +79,17 @@ func RegisterAndWaitForAllocs(f *framework.F, nomadClient *api.Client, jobFile s
 	require.Nil(err)
 	return allocs
 }
+
+func WaitForAllocRunning(t *testing.T, nomadClient *api.Client, allocID string) {
+	testutil.WaitForResultRetries(retries, func() (bool, error) {
+		defer time.Sleep(time.Millisecond * 100)
```

Comment:
There's no reason to defer this. Since you're overriding the testutil's retries and sleep time you may be better off with your own for loop anyway. The helper isn't doing much.

```suggestion
		time.Sleep(time.Millisecond * 100)
```